loadbalancer/algorithm.go

```diff
@@ -206,9 +207,12 @@ func (ch consistentHash) Less(i, j int) bool { return ch[i].hash < ch[j].hash }
 func (ch consistentHash) Swap(i, j int)      { ch[i], ch[j] = ch[j], ch[i] }
 
 func newConsistentHash(endpoints []string) routing.LBAlgorithm {
-	ch := consistentHash(make([]endpointHash, len(endpoints)))
+	const hashesPerEndpoint = 100
+	ch := consistentHash(make([]endpointHash, hashesPerEndpoint*len(endpoints)))
 	for i, ep := range endpoints {
-		ch[i] = endpointHash{i, hash(ep)}
+		for j := 0; j < hashesPerEndpoint; j++ {
+			ch[(hashesPerEndpoint*i)+j] = endpointHash{i, hash(fmt.Sprintf("%s-%d", ep, j))}
```

Comment:
I haven't tried to prevent collisions here, so in the event that two hashes point to the same endpoint, whichever one comes first after the sort will win

Reply:
I have found this article https://preshing.com/20110504/hash-collision-probabilities/ regarding collision probability estimation. Here is a collision probability list for `N` `int32` hashes:
```python
N = 2**32
probUnique = 1.0
for k in [10, 20, 100, 200, 1000, 2000, 10000]:
    probUnique = probUnique * (N - (k - 1)) / N
    print('{:5d} {:0.12f}'.format(k, 1 - probUnique))
```
```
   10 0.000000002095
   20 0.000000006519
  100 0.000000029569
  200 0.000000075903
 1000 0.000000308501
 2000 0.000000773929
10000 0.000003102001
```

Reply:
> I haven't tried to prevent collisions here

IMO its ok, we should not bother

Reply:
```python
def collision_prob(k, n):
    probUnique = 1.0
    for i in xrange(1, k):
        probUnique = probUnique * (n - (k - 1)) / n
    return 1 - probUnique


for k in [10, 20, 100, 200, 1000, 2000, 10000]:
    p = collision_prob(k, 2**32)
    print('{:5d} {:0.12f}'.format(k, p))
```
```
   10 0.000000018859
   20 0.000000084052
  100 0.000002281971
  200 0.000009220284
 1000 0.000232338248
 2000 0.000929959021
10000 0.023009582509
```

Reply:
Cool, good to have numbers for that, thanks :tada: